include/up-cpp/client/Notifier.h

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-FileType: SOURCE
+// SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef UP_CPP_CLIENT_NOTIFIER_H
+#define UP_CPP_CLIENT_NOTIFIER_H
+
+#include <uprotocol/v1/uattributes.pb.h>
+#include <uprotocol/v1/uri.pb.h>
+#include <uprotocol/v1/ustatus.pb.h>
+#include <up-cpp/datamodel/builder/UMessage.h>
+#include <up-cpp/transport/UTransport.h>
+
+#include <chrono>
+#include <memory>
+#include <optional>
+#include <tuple>
+#include <utility>
+
+namespace uprotocol::client {
+
+/// @brief Interface for uEntities to send a notification to a specified target
+///
+/// Like all L2 client APIs, the Notifier is a wrapper on top of the L1
+/// UTransport API; in this instance, it provides for the notification model.
+struct Notifier {
+	/// @brief Constructs a notifier connected to a given transport.
+	///
+	/// @post An internal UMessageBuilder will be configured based on the
+	///       provided attributes.
+	///
+	/// @param transport Transport to publish messages on.
+	/// @param source URI of this uE. If only the resource is set, the default
+	///               authority and entity from the transport will be
+	///               automatically filled in.
+	/// @param dest URI of the uE notifications will be sent to.
+	/// @param priority All pubhished messages will be assigned this priority.
+	/// @param ttl How long messages will be valid from the time publish() is
+	///            called.
+	Notifier(std::shared_ptr<transport::UTransport> transport,
+	         const v1::UUri& source, const v1::UUri& dest,
+	         std::optional<v1::UPriority> priority = {},
+	         std::optional<std::chrono::milliseconds> ttl = {});
+
+	using ListenHandle = transport::UTransport::ListenHandle;
+	using ListenCallback = transport::UTransport::ListenCallback;
+
+	/// @brief Register a callback to receive notifications.
+	///
+	/// @param source URI of the uE notifications will be received from.
+	/// @param dest URI of this uE. If only the resource is set, the default
+	///             authority and entity from the transport will be
+	///             automatically filled in.
+	[[nodiscard]] static std::tuple<v1::UStatus, ListenHandle> listen(
```

Comment:
I'd like to learn a little more about why `listen()` is in here vs being in its own class as it feels like you struck a nice balance with the other APIs of e.g. `Publisher` & `Subscriber`.

Reply:
Yeahhh... I couldn't decide between this model, where listen is just a static interface inside the sending half, and what pub/sub does where the Subscribe side is just a single function hanging out in a namespace.

I'd really like to do one or the other. If you can think of a good name for the "to be notified" half of notifications, I'll move the function there.

Reply:
I suppose I would as an end user not expect the listen portion to be in here when writing a uE for the first time. I'm sure they'd get used to it, but. Hmm 🤔 

What about NotificationTarget or NotificationClient?